### ISSUE TITLE
Announce new error messages in the error summary

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4795]
The content of the error summary should be marked as `aria-live="polite"`, so screen readers can tell users about new error messages showing there.

[+1 from customer](https://3.basecamp.com/3600924/buckets/2192969/messages/3524064701)

**Comment by avernet** [comment]
In repeats, the screen reader will read then last error added. For instance, if you already have the errors below, insert a row at the beginning, and make it invalid, one hears "4. First name Missing or incorrect value", while in fact it was the "1. First name Missing or incorrect value" that was added, and the others were shifted, so in a sense what is being read isn't technically correct In another sense, we hear what a sighted user would see as being added to the error summary.

<img width="952" alt="image" src="https://user-images.githubusercontent.com/106961/110055265-88532b00-7d11-11eb-99b6-041c7dc3a07a.png">

**Comment by ebruchez** [comment]
To do this right we'd need to be able to identify which are the new iterations added.

`ControlsComparator`, if the size of the repeat is different, does a full update to add the new iterations explicitly in the end. But it has access to the actual repeat iterations, and we could compare iterations based on node identity to be smarter about additions, and support inserting the new iterations in places other than the end.

In fact we have (very old) issue #1274 to track this. One issue, besides handling the comparison, is that we address controls using "effective ids" which are currently expected to match the position in the repeat.
